### ISSUE TITLE
Refactor/Rethink Model outputs, wrt logits

### ISSUE THREAD

**Issue by zphang** [jiant-dev/jiant#45]
(Issue referenced in code)

We abuse the name 'logits' in outputs for both `EmbeddingModel` and `RegressionModel`. Consider more outputs types, or rethinking our model output data structures.
